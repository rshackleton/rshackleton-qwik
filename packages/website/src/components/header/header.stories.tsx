import type { Meta, StoryObj } from 'storybook-framework-qwik';
import type { HeaderProps } from './header';
import { Header } from './header';

export default {
  title: 'Header',
  component: Header,
} as Meta<HeaderProps>;

export const Default: StoryObj<HeaderProps> = {};
