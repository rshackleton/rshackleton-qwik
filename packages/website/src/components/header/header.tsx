import { component$ } from '@builder.io/qwik';
import { Link } from '@builder.io/qwik-city';
import { Navigation } from '../navigation/navigation';

export type HeaderProps = {};

export const Header = component$<HeaderProps>(() => {
  return (
    <header class="flex flex-col items-center px-4 py-12 text-base text-neutral-800">
      <Link class="inline-block pb-4 text-xl font-bold underline-offset-4 hover:underline" href="/">
        Richard Shackleton
      </Link>
      <Navigation />
    </header>
  );
});
