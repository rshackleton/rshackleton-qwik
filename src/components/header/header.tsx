import { component$ } from '@builder.io/qwik';
import { Link } from '@builder.io/qwik-city';

export const Header = component$(() => {
  return (
    <header class="">
      <div class="site-container py-12">
        <Link class="text-lg underline underline-offset-2" href="/">
          rshackleton.co.uk
        </Link>
      </div>
    </header>
  );
});
