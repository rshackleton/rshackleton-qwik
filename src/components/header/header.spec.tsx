import { QwikCityMockProvider } from '@builder.io/qwik-city';
import { createDOM } from '@builder.io/qwik/testing';
import { expect, test } from 'vitest';
import { Header } from './header';

test(`[Header Component]: Renders without error`, async () => {
  const { screen, render } = await createDOM();
  await render(
    <QwikCityMockProvider>
      <Header />
    </QwikCityMockProvider>,
  );
  expect(screen.outerHTML).toMatchSnapshot();
});
