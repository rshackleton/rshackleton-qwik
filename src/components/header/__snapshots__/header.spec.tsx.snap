// Vitest Snapshot v1

exports[`[Header Component]: Renders without error 1`] = `"<host q:version=\\"0.16.2\\" q:container=\\"resumed\\" q:render=\\"dom-dev\\"><!--qv q:key=WmYC:bz_1 data-qrl=QwikCityMockProvider_component_WmYC5H00wtI--><!--qv q:key q:sref=0 q:s--><!--qv q:key=CedV:bz_0 data-qrl=Header_component_CedV26l7xxQ--><header><div class=\\"mx-auto max-w-screen-2xl px-8 py-8\\"><!--qv q:key=8gdL:SN_0 data-qrl=Link_component_8gdLBszqbaM--><a class=\\"text-lg underline underline-offset-2\\" href=\\"/\\" preventdefault:click=\\"\\"><!--qv q:key q:sref=2 q:s-->rshackleton.co.uk<!--/qv--></a><!--/qv--></div></header><!--/qv--><!--/qv--><!--/qv--></host>"`;
